server/terraform/modules/myhealth/main.tf

```diff
@@ -190,6 +190,52 @@ resource "google_compute_region_network_endpoint_group" "neg" {
   depends_on = [google_project_service.service]
 }
 
+
+# Google Cloud Armor security policy.
+resource "google_compute_security_policy" "policy" {
+  name = "lb-security-policy"
+
+  # IP Block list. 
+  rule {
+    action   = "deny(403)"
+    priority = "1000"
+    match {
+      versioned_expr = "SRC_IPS_V1"
+      config {
+        # Using an unused portion of the loopback block as a placeholder.
+        # Remove it once a real set of IP Ranges are present.
+        # Up to ten IP ranges can be created per rule.
+        src_ip_ranges = ["127.0.0.255/32","127.0.0.254"]
+      }
+    }
+    description = "Deny access to specified IPs"
+  }
+
+  rule {
+    action   = "allow"
+    priority = "2147483647"
+    match {
+      versioned_expr = "SRC_IPS_V1"
+      config {
+        src_ip_ranges = ["*"]
+      }
+    }
+    description = "default rule"
+  }
+
+  # In case  this resource is renamed in Terraform, the lifecycle must be:
+  #  1: Create new policy 
+  #  2: Configure backend to reference it
+  #  3: Delete old policy.
+  # To accomplish this, it is set to create_before_destroy, 
+  # then running apply twice completes the operation.
```

Comment:
Is the logic reversed here? Comment said "if resource is renamed, then lifecycle must be..."

I'm expecting "can't update in place, therefore need to rename when updating", I did this with the random suffix on the ssl certificate name:

https://github.com/WorldHealthOrganization/app/blob/master/server/terraform/modules/http-load-balancer/main.tf#L71-L84

Reply:
In what sense?
I could delete this, as no one really 'should' be renaming this resource from name = "lb-security-policy" to something else. And this is a bit of a distraction.

But what happens is that google_compute_backend_service.backend.security_policy references google_compute_security_policy.policy by name. So if you destroy nameOld and create nameNew, the steps need to be the three outlined above. Perhaps terraform has some way to describe this set of operations--it seems to know that google_compute_security_policy.policy needs to be ready to go for google_compute_backend_service.backend to be possible, but does not know that if the value is *changing* that it needs to be unreferenced first.
(Note: this could also be solved by having terraform unattach the 'old' security policy, then reattach it. That would mean updating google_compute_backend_service.backend twice, including having it be in a temporary state with no policy at all.)

Reply:
Ok, not much of an issue then